### PR TITLE
show default saved subtitle extension if it matches possibilities

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

**Comment by clsid2** [comment]
This uses indexes of two different lists? Might be error-prone if any of the two lists are modified.

Perhaps simplify it?
If ssa/ass -> use ass
Otherwise use srt?

Because when user saves subtitle, it may often be to convert it, and the common formats seem more logical as a default then.

**Comment by adipose** [comment]
Whatever you think is best.

We could possibly add a new option to "save a copy" vs "save/export as."

Maybe it's used to extract mkv subs for editing?

**Comment by clsid2** [comment]
Yes, it can also be used for embedded ones. But those are SRT or ASS/SSA anyway.

New option isn't needed.

**Comment by xLn2** [comment]
> new option to "save a copy" vs "save/export as."
> Maybe it's used to extract mkv subs for editing?

It might make sense to save the active internal/embedded subtitles from the same place without the need for a new option, multiple exports might be overkill as the normal user will be editing usually on a single language. 
It is probaby object of another issue topic, since it needs some work.

File: src/mpc-hc/MainFrm.cpp [hunk 1]
```diff
@@ -6331,19 +6331,19 @@ void CMainFrame::SubtitlesSave(const TCHAR* directory, bool silent)
...
+            filter += _T("SubStation Alpha (*.ssa)|*.ssa|"); //5
+            filter += _T("Advanced SubStation Alpha (*.ass)|*.ass|"); //6
             filter += _T("|");
 
             CSaveSubtitlesFileDialog fd(pRTS->m_encoding, m_pCAP->GetSubtitleDelay(), s.bSubSaveExternalStyleFile,
                                         _T("srt"), suggestedFileName, filter, types, GetModalParent());
 
-            if (pRTS->m_subtitleType >= 0 && size_t(pRTS->m_subtitleType) < types.size()) { //see Subtitle::GetSubtitleFileExt, but we have a limited set of extensions
-                fd.m_ofn.nFilterIndex = pRTS->m_subtitleType + 1; //nFilterIndex is 1-based
+            if (pRTS->m_subtitleType == 4 || pRTS->m_subtitleType == 5) {
```
Comment (clsid2) [inline src/mpc-hc/MainFrm.cpp]: `pRTS->m_subtitleType == Subtitle::SubType::SSA || pRTS->m_subtitleType == Subtitle::SubType::ASS`

Comment (adipose) [reply src/mpc-hc/MainFrm.cpp]: 🤦‍♂️

**Pull request merged**